src/hotspot/cpu/riscv/gc/z/z_riscv.ad

```diff
@@ -140,7 +140,7 @@ instruct zCompareAndSwapP(iRegINoSp res, indirect mem, iRegP oldval, iRegP newva
   ins_encode %{
     guarantee($mem$$disp == 0, "impossible encoding");
     Address ref_addr($mem$$Register);
-    z_color(_masm, this, $oldval_tmp$$Register, $oldval$$Register, t0);
+    z_color(_masm, this, $oldval_tmp$$Register, $oldval$$Register, t1);
```

Comment:
Here you can use newval_tmp instead t1, no?

Don't need to add additional regs.